### ISSUE TITLE
Bad port assignment in docker-compose.yml?

### ISSUE THREAD

**Issue by ed-5on2** [Chastilock/chastilock-backend#11]
https://github.com/Chastilock/chastilock-backend/blob/75285fbd3b6dad79e53ab3e8acb6e341f3f60d65/docker-compose.yml#L31

I had to change this line from:
13306:3306
to:
13306:13306
to get the migrations to run.

**Comment by ItsukiFox** [comment]
Hmm, that's weird! Usually the port there should not have any impact on the migrations you run, as long as you run them within the docker container, as described here: https://github.com/Chastilock/chastilock-backend#running-migrations-using-docker

The port forwarding system in docker works like this: you supply {HOST_PORT}:{CONTAINER_PORT}. Therefore the port on the host where mysql will be available is 13306, as described here: https://github.com/Chastilock/chastilock-backend#connecting-to-the-database-using-docker
But the container is listening on port 3306 (mysql default), and it is unchanged. So forwarding port 13306 to container port 13306 _should_ do nothing, as there should be no service running on the container port 13306. So I assume you did run the migrations outside of the docker container? But even then something is strange. Would you mind providing reproduction steps if we should look into it?

**Comment by SirHavok** [comment]
I just want to confirm I've not had this problem either.

Reproduction steps would be really handy

**Comment by ed-5on2** [comment]
I did a fresh clone of the repository, cd'd into the repository directory, and ran docker-compuse up -d with no problems.  At the same location, I then tried to run docker exec chastilock-backend npx sequelize-cli db:migrate
And received an error about port 3306.
I changed the docker-compose.yml file to 13306:13306 and re-ran without an error.

**Comment by LarsBaertschi** [comment]
Hmm, if you try the same thing again but don't change the port? I could suspect that maybe MYSQL was still starting up and that's the reason why you got the error message. Because I can't see how that change would do anything that would fix such an issue.

**Comment by ItsukiFox** [comment]
Hmm, I'm still not sure how one could have led to the other here. Maybe MYSQL was starting up, and it actually would have just needed time. So I don't think we should change anything there, unless you're able to reproduce it.

**Comment by ed-5on2** [comment]
I suspect it may have been a timing issue where I tried to do the migrations very quickly after starting the containers.  When I got an error message that referred to port 3303, I assumed it was probably the port number that was causing the error since the information immediately below in the readme mentioned port 13303.  After changing 3303 to 13303 in the docker-compose.yml, it succeeded, but could have just been that more time elapsed.  Unfortunately, I didn't copy the error message; live and learn.
